### PR TITLE
Release of version 4.3.3

### PR DESCRIPTION
**Pull request by tscheypidi**
### added
- **44_biodiversity** New biodiversity module. The realization bv_btc_mar21 now allows to calculate an area-based biodiversity value across all land types. Switch `c44_price_bv_loss` to implement cost for biodiversity loss.
- **15_food*** added 3 sigmoid food substitution scenarios
- **56_ghg_policy** Automatic sets for scenarios
- **60_bioenergy** Automatic sets for scenarios
- **70_livestock*** added 3 sigmoid feed substitution scenarios
- **scripts** added output script for disaggregation to GAINS regions
- **scripts** Automatic sets for 56_ghg_policy and 60_bioenergy
- **scripts** Added pre-commit hook

### fixed
- **60_bioenergy** Minimal bioenergy demand

### CODE REVIEW

File: core/sets.gms [hunk 1]
```diff
@@ -16,7 +16,8 @@
 
 sets
 
-   i all economic regions /CAZ,CHA,EUR,IND,JPN,LAM,MEA,NEU,OAS,REF,SSA,USA/
+   i all economic regions /
```
Comment (abhimishr) [inline core/sets.gms:19]: unnecessary whitespace can be removed

File: core/sets.gms [hunk 2]
```diff
@@ -57,10 +58,9 @@ sets
        OAS_151*OAS_172,
        REF_173*REF_179,
        SSA_180*SSA_190,
-       USA_191*USA_200/
+       USA_191*USA_200 /
```
Comment (abhimishr) [inline core/sets.gms:61]: same as above

File: core/sets.gms [hunk 3]
```diff
@@ -57,10 +58,9 @@ sets
...
 
-   cell(i,j) number of LPJ cells per region i
-      /
+   cell(i,j) number of LPJ cells per region i /
```
Comment (abhimishr) [inline core/sets.gms:63]: avoid whitespace changes

File: core/sets.gms [hunk 4]
```diff
@@ -72,11 +72,9 @@ sets
        OAS . OAS_151*OAS_172
        REF . REF_173*REF_179
        SSA . SSA_180*SSA_190
-       USA . USA_191*USA_200
-      /
+       USA . USA_191*USA_200 /
```
Comment (abhimishr) [inline core/sets.gms:75]: avoid whitespace changes

File: core/sets.gms [hunk 5]
```diff
@@ -72,11 +72,9 @@ sets
...
 
-   i_to_iso(i,iso) mapping regions to iso countries
-      /
+   i_to_iso(i,iso) mapping regions to iso countries /
```
Comment (abhimishr) [inline core/sets.gms:77]: avoid whitespace changes

File: core/sets.gms [hunk 6]
```diff
@@ -109,8 +107,8 @@ sets
        SSA . (MYT,NAM,NER,NGA,REU,RWA,SEN,SHN,SLE,SOM)
        SSA . (SSD,STP,SWZ,SYC,TCD,TGO,TZA,UGA,ZAF,ZMB)
        SSA . (ZWE)
-       USA . (USA)
-      /
+       USA . (USA) /
```
Comment (abhimishr) [inline core/sets.gms:110]: avoid whitespace changes

File: modules/20_processing/substitution_dec18/preloop.gms [hunk 7]
```diff
@@ -27,3 +32,5 @@ i20_processing_conversion_factors(t,"breeding","scp",kpr) = i20_scp_conversion_f
 *Therefore, the share of the respective feedstock is set to 1.
 i20_processing_shares(t_all,i,"scp",kpr) = 0;
 i20_processing_shares(t_all,i,"scp",kpr)$(i20_scp_conversion_factors(t_all,kpr) > 0) = 1;
+
```
Comment (abhimishr) [inline modules/20_processing/substitution_dec18/preloop.gms:35]: avoid whitespace changes

File: modules/44_biodiversity/bv_btc_mar21/declarations.gms [hunk 8]
```diff
@@ -0,0 +1,45 @@
+*** (C) 2008-2021 Potsdam Institute for Climate Impact Research (PIK),
+*** authors, and contributors see AUTHORS file
+*** This file is part of MAgPIE and licensed under GNU AGPL Version 3
+*** or later. See LICENSE file or go to http://www.gnu.org/licenses/
+*** Contact: magpie@pik-potsdam.de
+
+parameters
+ pc44_bv_weighted(j,landcover44)		        Current range rarity weighted bii (Mha)
+ p44_price_bv_loss(t)			                biodiversity value loss price factor (US$ per ha of bv loss)
```
Comment (abhimishr) [inline modules/44_biodiversity/bv_btc_mar21/declarations.gms:9]: Can you check again if we allow only USD or US$? I could not find instances of US$ except for one place in sticky implementation. USD is recognized correctly by goxygen for documentation

Comment (abhimishr) [reply modules/44_biodiversity/bv_btc_mar21/declarations.gms:9]: If you decide on USD i'd suggest changing all instances of US$ to USD (even in sticky)

File: modules/56_ghg_policy/price_jan20/sets.gms [hunk 9]
```diff
@@ -5,6 +5,71 @@
 *** |  MAgPIE License Exception, version 1.0 (see LICENSE file).
 *** |  Contact: magpie@pik-potsdam.de
 
+*###############################################################################
```
Comment (abhimishr) [inline modules/56_ghg_policy/price_jan20/sets.gms:8]: It comes from https://github.com/magpiemodel/magpie/commit/158661e37292db77c85f83a348331bee5e78cebd but why this special set writing to this file @flohump ?

**Review by abhimishr — CHANGES_REQUESTED** [review]
Seems fine in general. Some comments about whitespaces and the commits should be avoided if possible with those whitespeace changes. An additional question to @flohump based on https://github.com/magpiemodel/magpie/commit/158661e37292db77c85f83a348331bee5e78cebd

File: core/sets.gms (hunk 1 above)
Comment (tscheypidi) [reply core/sets.gms:19]: This part is created automatically

File: core/sets.gms (hunk 2 above)
Comment (tscheypidi) [reply core/sets.gms:61]: see above

File: core/sets.gms (hunk 3 above)
Comment (tscheypidi) [reply core/sets.gms:63]: see above

File: modules/56_ghg_policy/price_jan20/sets.gms (hunk 9 above)
Comment (flohump) [reply modules/56_ghg_policy/price_jan20/sets.gms:8]: These are automatic sets, which are generated based on the content of the file.

File: modules/44_biodiversity/bv_btc_mar21/declarations.gms (hunk 8 above)
Comment (flohump) [reply modules/44_biodiversity/bv_btc_mar21/declarations.gms:9]: @pvjeetze Could you check on this?

Comment (flohump) [reply modules/44_biodiversity/bv_btc_mar21/declarations.gms:9]: @tscheypidi Changes should be made directly in the rc branch?

Comment (tscheypidi) [reply modules/44_biodiversity/bv_btc_mar21/declarations.gms:9]: yes

Comment (pvjeetze) [reply modules/44_biodiversity/bv_btc_mar21/declarations.gms]: I've changed the units to USD in https://github.com/magpiemodel/magpie/pull/266/commits/ec099f61dfd757623df34537a8429c88f317885b

File: core/sets.gms (hunk 5 above)
Comment (flohump) [reply core/sets.gms:77]: These sets are created automatically. The function that writes these sets has been changed slightly.

File: core/sets.gms (hunk 6 above)
Comment (flohump) [reply core/sets.gms:110]: These sets are created automatically. The function that writes these sets has been changed slightly.

File: modules/20_processing/substitution_dec18/preloop.gms (hunk 7 above)
Comment (flohump) [reply modules/20_processing/substitution_dec18/preloop.gms:35]: One additional line at the end is not critical.

File: core/sets.gms (hunk 4 above)
Comment (flohump) [reply core/sets.gms:75]: These sets are created automatically. The function that writes these sets has been changed slightly.

File: core/sets.gms (hunk 3 above)
Comment (flohump) [reply core/sets.gms:63]: These sets are created automatically. The function that writes these sets has been changed slightly.

File: core/sets.gms (hunk 2 above)
Comment (flohump) [reply core/sets.gms:61]: These sets are created automatically. The function that writes these sets has been changed slightly.

File: core/sets.gms (hunk 1 above)
Comment (flohump) [reply core/sets.gms:19]: These sets are created automatically. The function that writes these sets has been changed slightly.

**Review by flohump — APPROVED** [review]
Seems all fine to me.

**Review by abhimishr — APPROVED** [review]
🚀 Seems fine!
I noticed a typo I introduced. Can you maybe fix this in the commits here 
`*' @title Age class` in module 29 `module.gms`. I committed it as "production" but it should be Age class

**Pull request merged**